plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -165,6 +300,16 @@ public CompletableFuture<Collection<Slice>> finish()
         fileDeletions.forEach((path, deletion) ->
                 fragments.addAll(rewriteFile(new Path(path.toStringUtf8()), deletion)));
 
+        if (cdfEnabled && cdfPageSink != null) {
+            cdfPageSink.finish().join().stream()
```

Comment:
> .join()

this is right thing to do only because the future is known to be completed already
otherwise you would want wrap it in a completion closure
use `io.airlift.concurrent.MoreFutures#getDone` to indicate that

`insertPageSink.finish().join()` above can be fixed in a separate PR

Reply:
I will use that function, but I am not sure I understand. 
Why would i want it to wrap ? `finish()` returns a `CompletableFuture` and join will simply make current thread wait for that to complete. Why is that bad ?

Reply:
Why would we block on an incomplete Future in a function that can simply return a future?
we would rather transform the result lazily, without blocking